pylint/extensions/emptystring.py

```diff
@@ -41,31 +33,45 @@ class CompareToEmptyStringChecker(checkers.BaseChecker):
 
     @utils.only_required_for_messages("compare-to-empty-string")
     def visit_compare(self, node: nodes.Compare) -> None:
-        _operators = ["!=", "==", "is not", "is"]
-        # note: astroid.Compare has the left most operand in node.left
-        # while the rest are a list of tuples in node.ops
-        # the format of the tuple is ('compare operator sign', node)
-        # here we squash everything into `ops` to make it easier for processing later
+        """Checks for comparisons to empty string.
+
+        Most of the time you should use the fact that empty strings are false.
+        An exception to this rule is when an empty string value is allowed in the program
+        and has a different meaning than None!
+        """
+        _operators = {"!=", "==", "is not", "is"}
+        # note: astroid.Compare has the left most operand in node.left while the rest
+        # are a list of tuples in node.ops the format of the tuple is
+        # ('compare operator sign', node) here we squash everything into `ops`
+        # to make it easier for processing later
         ops: list[tuple[str, nodes.NodeNG | None]] = [("", node.left)]
         ops.extend(node.ops)
         iter_ops = iter(ops)
         ops = list(itertools.chain(*iter_ops))  # type: ignore[arg-type]
-
         for ops_idx in range(len(ops) - 2):
             op_1: nodes.NodeNG | None = ops[ops_idx]
             op_2: str = ops[ops_idx + 1]  # type: ignore[assignment]
             op_3: nodes.NodeNG | None = ops[ops_idx + 2]
             error_detected = False
-
+            if op_1 is None or op_3 is None or op_2 not in _operators:
+                continue
+            node_name = ""
             # x ?? ""
-            if utils.is_empty_str_literal(op_1) and op_2 in _operators:
+            if utils.is_empty_str_literal(op_1):
                 error_detected = True
```

Comment:
Are these `error_detected` values necessary; could we instead rely on `if node_name` further down?

Reply:
Of course the empty string case :D